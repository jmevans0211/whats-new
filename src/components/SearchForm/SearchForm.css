header {
  border-bottom: black solid 5px;
  display: flex;
  justify-content: space-between;
  background-image: url('./header_background.jpg');
}

h1 {
  background-color: red;
  color: white;
  font-size: 70px;
  margin-left: 60px;
  padding: 0 22px 6px 23px;
}

/* h1 span {
  color: #11CEE3;
} */

form {
  height: 60px;
  padding: 20px;
  width: 300px;
}

input, button {
  height: 100%;
  width: 50%;
}

button {
  color: white;
  cursor: pointer;
  font-weight: 500;
  background-color: red
}