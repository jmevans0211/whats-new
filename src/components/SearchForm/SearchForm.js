import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchField: ''
    }
  };

  handleChange = (event) => {
    this.setState({searchField: event.target.value})
  }

  handleSearch = (event) => {
    event.preventDefault()
    console.log('in handleSearch***', this.state.searchField)
    
  }

  render() {
    return (
      <form>
        <input type="text" placeholder="Search" onChange={event => this.handleChange(event)}/>
        <button onClick={this.handleSearch}>Submit Search</button>
      </form>
    )
  }
}

export default SearchForm;