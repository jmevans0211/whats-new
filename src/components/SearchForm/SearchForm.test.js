import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {
  it('should match the snapshot', () => {
    const mockfilterSearch = jest.fn()
    const wrapper = shallow(<SearchForm  search={mockfilterSearch}/>)

    expect(wrapper).toMatchSnapshot();
  });



  // it('should match the snapshot', () => {
  //   const wrapper = shallow(<Form addIdea={mockAddIdea} />)
  //   expect(wrapper).toMatchSnapshot();
  // });

}); //<<-----end of describe block


//TESTS TO HAVE
// state
  //empty search field

//handle change
  //test that the state has indeed changed when there is an input
    //mimic input
      //set state
    
//onClick
  //that handle search has been called