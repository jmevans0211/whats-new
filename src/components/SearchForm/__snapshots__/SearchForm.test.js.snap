// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Submit Search
  </button>
</form>
`;
